File: Programmers/Level4/hopscotch(pjct)/Hopscotch.java
class Hopscotch {
    int hopscotch(int[][] board, int size) {
        int result = -12345689;

        for(int i=1; i<board.length; ++i)
        {
            for(int j=0; j<board[0].length; ++j)
            {
                int maxNum = -123456789;
                for(int k=0; k<board[0].length; ++k)
                {
                    if(k==j)
                    {
                        continue;
                    }
                    if(maxNum < board[i-1][k])
                    {
                        maxNum = board[i-1][k];
                    }
                }
                board[i][j]+=maxNum;
            }
        }

        for(int i=0; i<board[0].length; ++i)
        {
            if(result < board[board.length-1][i])
            {
                result = board[board.length-1][i];
            }
        }

        return result;
    }

    public static void main(String[] args) {
        Hopscotch c = new Hopscotch();
        int[][] test = { { 1, 2, 3, 5 }, { 5, 6, 7, 8 }, { 4, 3, 2, 1 } };
        System.out.println(c.hopscotch(test, 3));
    }
}

File: Programmers/Level4/hopscotch(pjct)/hopscotch.py
def hopscotch(board, size):
    def dfs(f,p):
        max_val = -1
        if size == f:
            return 0
        for i in range(len(board[0])):
            if i == p:
                continue
            max_val = max(max_val,board[f][i]+dfs(f+1,i))
        return max_val
    return dfs(0,-1)

board =  [[ 1, 2, 3, 5 ], [ 5, 6, 7, 8 ], [4, 3, 2, 1]]
print(hopscotch(board, 3))